src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,280 @@
+        this.inputList.add(task);
+    }
+
+    /**
+     * Returns The task inside the task list at the index specified
+     *
+     * @param index index of task inside the tasklist
+     * @return A task
+     */
+    public Task get(int index) {
+        return this.inputList.get(index);
+    }
+
+    /**
+     * Returns a ToDoTask based on the user input and adds it to the list
+     *
+     * @param action  The string action input by the user
+     * @return A ToDoTask
+     */
+    public ToDoTask handleToDoTask(String action) {
+
+        int index = action.indexOf(" ");
+        String description = action.substring(index + 1);
+
+        ToDoTask toDoTask = new ToDoTask(description);
+
+        this.add(toDoTask);
+
+        return toDoTask;
+    }
+    /**
+     * Returns a EventTask based on the user input and adds it to the list
+     *
+     * @param action  The string action input by the user
+     * @return A EventTask
+     */
+    public EventTask handleEventTask(String action) {
+
+        int actionIndex = action.indexOf(" ");
+        int descriptionIndex = action.indexOf("/");
+
+        String description = action.substring(actionIndex + 1, descriptionIndex - 1);
+        String event = action.substring(descriptionIndex + 4);
+
+        EventTask eventTask = new EventTask(description, event);
+
+        this.add(eventTask);
+
+        return eventTask;
+    }
+
+    /**
+     * Returns a DeadLineTask based on the user input and adds it to the list
+     *
+     * @param action  The string action input by the user
+     * @return A DeadLineTask
+     */
+    public DeadlineTask handleDeadlineTask(String action) {
+        int actionIndex = action.indexOf(" ");
+        int descriptionIndex = action.indexOf("/");
+        
+        String description = action.substring(actionIndex + 1, descriptionIndex - 1);
+        String deadline = action.substring(descriptionIndex + 4);
+
+        DeadlineTask deadlineTask = new DeadlineTask(description, deadline);
+
+        this.add(deadlineTask);
+
+        return deadlineTask;
+    }
+
+    /**
+     * Returns A task that is mark as done
+     * The task at the specified index is
+     * marked as done before being returned
+     *
+     * @param index index of task in list
+     * @return Task that is marked
+     */
+    public Task handleDone(int index) {
+        Task markDone = this.inputList.get(index - 1);
+
+        markDone.markAsDone();
+
+        return markDone;
+    }
+
+    /**
+     * Returns the task that was at specified index is now removed.
+     *
+     * @param index index of Task in list
+     * @return The task that is removed
+     */
+    public Task handleDelete(int index) {
+        Task task = this.inputList.remove(index - 1);
+
+        return task;
+    }
+
+    /**
+     * Returns The String that will be used to write into the file.
+     *
+     * @return A string to be written into the file.
+     */
+    public String getListToWrite() {

```

Comment:
getListToString seems more appropriate here as you are returning a String as the output.

Reply:
Thanks, getListToString does sound nicer :D